torch/nn/modules/module.py

```diff
@@ -142,8 +163,12 @@ def add_module(self, name, module):
         if not isinstance(module, Module) and module is not None:
             raise TypeError("{} is not a Module subclass".format(
                 torch.typename(module)))
-        if hasattr(self, name) and name not in self._modules:
+        elif hasattr(self, name) and name not in self._modules:
             raise KeyError("attribute '{}' already exists".format(name))
+        elif '.' in name:
+            raise KeyError("module name can't contain \".\"")
```

Comment:
This breaks the densenet's implementation in torchvision. See the error at https://travis-ci.org/lanpa/tensorboard-pytorch/jobs/369432497